cli/Sources/TuistCache/Services/LoadModuleCacheService.swift

```diff
@@ -0,0 +1,100 @@
+import Foundation
+import Mockable
+import OpenAPIRuntime
+import OpenAPIURLSession
+import TuistHTTP
+import TuistServer
+
+@Mockable
+public protocol LoadModuleCacheServicing: Sendable {
+    func loadModuleCache(
+        fullHandle: String,
+        hash: String,
+        name: String,
+        cacheCategory: String,
+        serverURL: URL,
+        authenticationURL: URL,
+        serverAuthenticationController: ServerAuthenticationControlling
+    ) async throws -> Data
+}
+
+public enum LoadModuleCacheServiceError: LocalizedError {
+    case unknownError(Int)
+    case unauthorized(String)
+    case forbidden(String)
+    case notFound(String)
+    case badRequest(String)
+
+    public var errorDescription: String? {
+        switch self {
+        case let .unknownError(statusCode):
+            return "The module cache artifact could not be downloaded due to an unknown response of \(statusCode)."
+        case let .unauthorized(message),
+             let .forbidden(message),
+             let .notFound(message),
+             let .badRequest(message):
+            return message
+        }
+    }
+}
+
+public final class LoadModuleCacheService: LoadModuleCacheServicing {
```

Comment:
```suggestion
public struct LoadModuleCacheService: LoadModuleCacheServicing {
```

we lean to using structs over classes nowadays